glue/viewers/image/state.py

```diff
@@ -543,8 +543,15 @@ def __init__(self, layer=None, viewer_state=None, **kwargs):
 
         self.update_from_dict(kwargs)
 
+        if isinstance(self.preferred_cmap, str):
+
+            for i, element in enumerate(colormaps.members):
+                if element[0] == self.preferred_cmap:
+                    self.preferred_cmap = element[1]
+
         if self.cmap is None:
-            self.cmap = colormaps.members[0][1]
+            self.cmap = self.preferred_cmap if self.preferred_cmap\
```

Comment:
Rather than storing ``preferred_cmap`` on the state, why not just access ``self.layer.style.preferred_cmap`` here? (this will avoid ``self.layer.style.preferred_cmap`` and ``self.preferred_cmap`` from getting out of sync.

Reply:
Done ✅Double-checked that the code still works after modifying.